src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -1148,3 +1164,172 @@ const isMobileFieldUpdateAllowed = (
     },
   )
 }
+
+export const createTwilioCredentials = (
+  twilioCredentials: TwilioCredentials,
+  form: IPopulatedForm,
+): ResultAsync<unknown, ApplicationError> => {
+  if (!isCredentialsValid(twilioCredentials))
+    return errAsync(new MalformedParametersError('Credentials are invalid!'))
+
+  const twilioCredentialsData: TwilioCredentialsData =
+    new TwilioCredentialsData(twilioCredentials)
+
+  const msgSrvcName = `formsg/${process.env.SSM_PREFIX}/form/${form._id}/twilio`
+  const body: SecretsManager.Types.CreateSecretRequest = {
+    Name: msgSrvcName,
+    SecretString: twilioCredentialsData.toString(),
+  }
+  logger.info({
+    message: `no msgSrvcName, creating Twilio credentials for form ${form._id}`,
+    meta: {
+      action: 'createTwilioCredentials',
+      formId: form._id.toHexString(),
+      msgSrvcName,
+    },
+  })
+
+  return ResultAsync.fromPromise(
+    mongoose.connection.transaction(async (session: ClientSession) => {
```

Comment:
Actually, I was looking at this part of the official doc to perform the transaction, which I felt was cleaner
https://mongoosejs.com/docs/api/connection.html#connection_Connection-transaction.

Do you still want me to switch to the other syntax?

Reply:
That's probably not the ideal way to perform transactions in mongoose, since it's not in the transaction docs: https://mongoosejs.com/docs/transactions.html.

Please switch to the other syntax, I'm not that keen on the passing of sessions